src/PowerShellEditorServices/Services/PowerShellContext/Session/PSReadLinePromptContext.cs

```diff
@@ -98,7 +80,9 @@ internal class PSReadLinePromptContext : IPromptContext
             {
                 pwsh.Runspace = runspace;
                 var psReadLineType = pwsh
-                    .AddScript(ReadLineInitScript, useLocalScope: true)
+                    .AddCommand("Import-Module")
```

Comment:
We can't assume that Import-Module will be available in all runspaces. I think we should centralize all module loading to PowerShellContextService.Create();

If you check out my changes in https://github.com/PowerShell/PowerShellEditorServices/pull/1507 you'll see that I conditionally load modules using either Import-Module or InitialSessionState.ImportPSModulesFromPath depending on whether or not the runspace has a FileSystem provider. We already know at that point whether or not PSReadline will be required based on hostStartupInfo.ConsoleReplEnabled, so I say we load it there instead.

Reply:
On further inspection, I realized that this PowerShell instance is temporary for getting the type information for the proxy so whatever I load with PowerShellContextService.Create(); won't show up here. Carry on.

Reply:
Turns out you guys are assigning the runspace to the newly created PowerShell, therefore it _is_ inheriting the language mode/modules so I stand by my initial assertion of centrally locating all modules.

In case it is unclear, in order to load modules that require file system access into a constrained runspace that has no FileSystem provider, you have to use InitialSessionState.ImportPSModulesFromPath, Import-Module won't work.

Reply:
Also, are you sure this actually worked? 
![image](https://user-images.githubusercontent.com/1424395/123521394-f1253100-d67b-11eb-8d4c-2cef7d8b857e.png)

Reply:
Here's how I think we should do it:

```c#
var assemblies = AppDomain.CurrentDomain.GetAssemblies();
                var psReadLineType = assemblies.FirstOrDefault(a => a.FullName.Contains("Microsoft.PowerShell.PSReadLine2"))?.ExportedTypes?.FirstOrDefault(a => a.Name == "PSConsoleReadLine");

                if (psReadLineType == null)
                {
                    logger.LogWarning("PSReadLine unable to be loaded: {Reason}", pwsh.HadErrors ? pwsh.Streams.Error[0].ToString() : "<Unknown reason>");
                    return false;
                }

                try
                {
                    readLineProxy = new PSReadLineProxy(psReadLineType, logger);
                }
                catch (InvalidOperationException e)
                {
                    // The Type we got back from PowerShell doesn't have the members we expected.
                    // Could be an older version, a custom build, or something a newer version with
                    // breaking changes.
                    logger.LogWarning("PSReadLine unable to be loaded: {Reason}", e);
                    return false;
                }
```

Reply:
Actually, now that we're not using PowerShell to resolve it, we don't need to pass runspace in anymore at all, it could be

```c#
internal static bool TryGetPSReadLineProxy(
    ILogger logger,
    out PSReadLineProxy readLineProxy)
{
    readLineProxy = null;
    logger.LogTrace("Attempting to load PSReadLine");
    var psReadLineAssembly = AppDomain.CurrentDomain.GetAssemblies().FirstOrDefault(a => a.FullName.Contains("Microsoft.PowerShell.PSReadLine2"));
    if(psReadLineAssembly is null)
    {
        logger.LogWarning("Microsoft.PowerShell.PSReadLine2 not found in loaded assemblies");
        return false;
    }
    var psReadLineType = psReadLineAssembly?.ExportedTypes?.FirstOrDefault(a => a.Name == "PSConsoleReadLine");

    if (psReadLineType == null)
    {
        logger.LogWarning("PSConsoleReadLine type not found in Microsoft.PowerShell.PSReadLine2");
        return false;
    }

    try
    {
        readLineProxy = new PSReadLineProxy(psReadLineType, logger);
    }
    catch (InvalidOperationException e)
    {
        // The Type we got back from PowerShell doesn't have the members we expected.
        // Could be an older version, a custom build, or something a newer version with
        // breaking changes.
        logger.LogWarning("PSReadLine unable to be loaded: {Reason}", e);
        return false;
    }

    return true;
}
```

Reply:
> Also, are you sure this actually worked?

No I'm not, I only know the tests passed. I needed to put this through some manual testing still 😆

Reply:
I found a few other rough spots in my testing over the weekend, I have my own pull request with the changes:
https://github.com/PowerShell/PowerShellEditorServices/pull/1516

Turns out we weren't at all honoring the path provided by the -BundledModulesPath param

Reply:
Ah you already found it! Now if I can only figure out why the tests never pass when I create pull requests...

Reply:
Just trying to get the assembly itself will fail since it's not yet loaded. Also PSES surfaces PSReadLine commands in the terminal, so we have to import the module properly.